lib/OpenQA/JobSettings.pm

```diff
@@ -17,6 +17,30 @@ package OpenQA::JobSettings;
 use strict;
 use warnings;
 
+sub generate_settings {
+    my ($params) = @_;
+    my $settings = $params->{settings};
+    my @worker_class;
+    for my $entity (qw (product machine test_suite job_template)) {
+        next unless $params->{$entity};
+        my @entity_settings = $params->{$entity}->settings;
+        for my $setting (@entity_settings) {
+            if ($setting->key eq 'WORKER_CLASS') {
+                push @worker_class, $setting->value;
+                next;
+            }
+            $settings->{$setting->key} = $setting->value;
+        }
+        $settings->{BACKEND} = $params->{$entity}->backend if ($entity eq 'machine');
```

Comment:
Where is this coming from? I don't see where that maps to the old code.

Reply:
good catch

Reply:
In the old code, when creating a job using `jobs post`, the `BACKEND` is not added to the settings. (I think it's a bug in the old logic. BTW it's also the answer to the comment you added in `04-jobs.t`).  In the `schedule_iso`, the `BACKEND` is added to the settings. it is done is https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Schema/Result/ScheduledProducts.pm#L543 . So I added this to settings when using `jobs post`.

Reply:
That is what I was missing. So it looks like you found a bug afterall. Sweet! Thanks for confirming!